### DISCUSSION TITLE
기획 문서에 관한 질문들 (은행 기능 관련)

### DISCUSSION THREAD

**Discussion by mille-feu** [Tiny-Metro/Poly-Metro#51] [Q&A]
- 통계 기능으로 볼 수 있는 "이용 승객 수"는 목적지에 도착한 승객을 의미하는지
- 대출 기능의 상세설명 필요
  - 자동 상환은 강제적으로 상환되는 금액인지
  - 이자는 복리인지
  - 자동 상환 기능에 의해 소지금이 음수가 될 수 있는지, 패널티는 있는지
  - 자동 상환의 시간 기준은 대출 시점부터인지, 게임 시간 기준 0시인지
- 투자 미션은 시작부터 전부 해금되어있는지

**Comment by seong0929** [comment]
- 통계 기능의 "이용 승객 수"는 도착한 승객을 기준으로 count
- 대출
   - 대출 상환액 계산 식: M=P*(J/(1-(1+J)-N)) (M: 월 상환액, P: 원금(총 대여금), J: 유효이자율, N: 상환 횟수) -> 댓글 식으로 수정
   - 소지금 음수는 가능하지만 음수인채로 오래 지속 될 시 게임 오버(문서 정리되는 대로 수정하겠습니다.)
   - 자동 상환 기준: 게임 내 일주일이 끝나는 0시
- 투자미션은 시작 시 하나의 투자 미션이 존재, 1주일마다 새로운 투자 미션 생성, 최대 3개의 투자 미션을 가질 수 있다. 3개가 꽉 찬 상태에서 투자 미션 생성 시 수락한 투자를 제외한 가장 오래된 투자 미션부터 삭제, refesh버튼 누르면 1주일이 안 지나도 새로운 투자 미션 등장(원래 있던 투자 미션 중 받지 않은 것 외에 모두 refresh)

**Reply by cstrnull00** [reply]
* 대출 상환액 관련 수정사항 
  * 원리금 균등상환
  * $W=\frac{P\times (1+\frac{R}{52})^T}{T}$
    * W : 주별 상환액
    * P : 원금
    * R : 이자 (연 이자율)
    * T : 상환 기간
 * 대출은 주마다 상환

**Reply by mille-feu** [reply]
투자미션 리프레시는 패널티나 조건같은게 없나요?

**Reply by mille-feu** [reply]
그리고, 자동상환과 주마다 상환이 혼동되는데 조금 자세한 설명 부탁드립니다.
추가로, 대출 기한에 대한 정보가 없습니다.

**Reply by cstrnull00** [reply]
> 투자미션 리프레시는 패널티나 조건같은게 없나요?

투자 미션 refresh는 1주일에 한 번만 가능하게 하려고 합니다. 그 외 조건은 없습니다.

**Reply by cstrnull00** [reply]
> 그리고, 자동상환과 주마다 상환이 혼동되는데 조금 자세한 설명 부탁드립니다. 추가로, 대출 기한에 대한 정보가 없습니다.

그냥 모든 대출에 상관 없이 위의 식을 따르고, 대출 기간 기준 한 주가 지나면 저 금액이 빠져나간다고 보시면 될 것 같습니다.
대출 기한에 대한 정보는 이후 업데이트될 예정입니다. 일단 지금 버전에서는 전부 20주로 통일해주시면 될 것 같습니다.

**Reply by mille-feu** [reply]
> > 투자미션 리프레시는 패널티나 조건같은게 없나요?
> 
> 투자 미션 refresh는 1주일에 한 번만 가능하게 하려고 합니다. 그 외 조건은 없습니다.

리프레시 되는 미션의 갯수는 현재 가질 수 있는 미션의 갯수와 같게 하면 되는지, 보유 미션 개수가 3개 미만일때는 리프레시가 불가하게 할지에 대한 의견 바랍니다

**Reply by cstrnull00** [reply]
> > > 투자미션 리프레시는 패널티나 조건같은게 없나요?
> > 
> > 
> > 투자 미션 refresh는 1주일에 한 번만 가능하게 하려고 합니다. 그 외 조건은 없습니다.
> 
> 리프레시 되는 미션의 갯수는 현재 가질 수 있는 미션의 갯수와 같게 하면 되는지, 보유 미션 개수가 3개 미만일때는 리프레시가 불가하게 할지에 대한 의견 바랍니다

보유 미션이 3개 미만일 때 역시 리프레시가 가능하며, 리프레시되는 미션의 개수 역시 현재 가질 수 있는 미션의 수와 같게 하면 됩니다.
모든 경우에서 현재 수락 중인 미션은 바뀌지 않아야 합니다.

**Reply by mille-feu** [reply]
> > > > 투자미션 리프레시는 패널티나 조건같은게 없나요?
> > > 
> > > 
> > > 투자 미션 refresh는 1주일에 한 번만 가능하게 하려고 합니다. 그 외 조건은 없습니다.
> > 
> > 
> > 리프레시 되는 미션의 갯수는 현재 가질 수 있는 미션의 갯수와 같게 하면 되는지, 보유 미션 개수가 3개 미만일때는 리프레시가 불가하게 할지에 대한 의견 바랍니다
> 
> 보유 미션이 3개 미만일 때 역시 리프레시가 가능하며, 리프레시되는 미션의 개수 역시 현재 가질 수 있는 미션의 수와 같게 하면 됩니다. 모든 경우에서 현재 수락 중인 미션은 바뀌지 않아야 합니다.

수락 가능 미션 슬롯이랑 수락중인 미션 슬롯은 별도인가요
EX)
- 은행에 미션 A B C가 있음, 미션 A를 수락 시
  - 은행에 미션 A가 사라지고 B C D로 갱신, A는 다른 UI로 확인가능
  - 은행에 미션 A B C가 있음, A는 수락중으로 표시됨, 리프레시할 경우 은행에 미션 A D E 가 표시됨
어느쪽이 맞는 해석일까요

**Reply by cstrnull00** [reply]
> > > > > 투자미션 리프레시는 패널티나 조건같은게 없나요?
> > > > 
> > > > 
> > > > 투자 미션 refresh는 1주일에 한 번만 가능하게 하려고 합니다. 그 외 조건은 없습니다.
> > > 
> > > 
> > > 리프레시 되는 미션의 갯수는 현재 가질 수 있는 미션의 갯수와 같게 하면 되는지, 보유 미션 개수가 3개 미만일때는 리프레시가 불가하게 할지에 대한 의견 바랍니다
> > 
> > 
> > 보유 미션이 3개 미만일 때 역시 리프레시가 가능하며, 리프레시되는 미션의 개수 역시 현재 가질 수 있는 미션의 수와 같게 하면 됩니다. 모든 경우에서 현재 수락 중인 미션은 바뀌지 않아야 합니다.
> 
> 수락 가능 미션 슬롯이랑 수락중인 미션 슬롯은 별도인가요 EX)
> 
> * 은행에 미션 A B C가 있음, 미션 A를 수락 시
>   
>   * 은행에 미션 A가 사라지고 B C D로 갱신, A는 다른 UI로 확인가능
>   * 은행에 미션 A B C가 있음, A는 수락중으로 표시됨, 리프레시할 경우 은행에 미션 A D E 가 표시됨
>     어느쪽이 맞는 해석일까요

후자입니다.
